### PR TITLE
Data Architecture 2.0: Checkout Confirmation

### PR DESCRIPTION
**Pull request by jeffkamo**
**JIRA**: [WEB-1113](https://mobify.atlassian.net/browse/WEB-1113)

## Changes
- Fetch data from confirmation page, or from redux store based on data from previous steps
- Add functionality to the confirmation page's "Registration" form
    - Registers the user
    - Updates the user's address book with information received in previous steps

## Todos
- [x] Create actions, et al. for submitting Registration form
    - [x] Hide registration form on success
    - [x] After successful registration, Log user in
- ~~Add "on mount" login detection (should persist from Shipping/Payment)~~
- [x] Retrieve order # from page
- [x] Retrieve order URL (if logged in)

## Post _"Integration with Shipping & Payment"_ Todo
- [x] Fix router path to `checkout/onepage/success/` to enable proper fetching of content
- [x] Verify that when logged out, arriving on the Confirmation Page also correctly resets the Cart count to zero

## How to test-drive this PR
- Run `npm run dev`
- Preview [Merlin's Potions](https://preview.mobify.com/?url=https%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1)
- add an item to the cart
- proceed through the checkout to the confirmation page
- confirm the confirmation page works as expected for logged in users and guest users

### CODE REVIEW

File: web/app/styles/utilities/_text.scss [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 .u-text-all-caps {
     @include text-all-caps(true);
+
+    letter-spacing: initial !important; // @TODO: maybe remove letter-spacing from text-all-caps() method in Spline
```
Comment (mlworks) [inline web/app/styles/utilities/_text.scss]: Agreed, I've experienced same issue in Crabtree build.

File: web/app/containers/checkout-confirmation/parsers/checkout-confirmation.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+const checkoutConfirmationParser = ($, $html) => {
```
Comment (fractaltheory) [inline web/app/containers/checkout-confirmation/parsers/checkout-confirmation.js]: Is this file required anymore?

File: web/app/containers/checkout-confirmation/partials/checkout-confirmation-form.jsx [hunk 3]
```diff
@@ -49,20 +57,54 @@ CheckoutConfirmationForm.propTypes = {
...
+    submitRegistrationForm: React.PropTypes.func,
+
+    /**
+     * Redux-form internal
+     */
     submitting: React.PropTypes.bool
 }
 
 const validate = (values) => {
     const errors = {}
```
Comment (fractaltheory) [inline web/app/containers/checkout-confirmation/partials/checkout-confirmation-form.jsx]: I don't get an error if I try to submit an empty form, also I don't see any validation for ensuring there's a number in the submitted password value (the help text for this form says it should have a number)

File: web/app/containers/checkout-confirmation/actions.js [hunk 4]
```diff
@@ -21,3 +56,139 @@ export const fetchContents = () => {
...
+}
+
+export const initiateBillingAndShippingUpdate = () => {
+    return (dispatch) => {
+        // Grab required form data in prep for updating shipping/billing information
+        const editCustomerAddressURL = 'https://www.merlinspotions.com/customer/address/edit/'
+        makeRequest(editCustomerAddressURL)
+            .then(jqueryResponse)
+            .then((res) => {
+                const [$, $response] = res // eslint-disable-line no-unused-vars
```
Comment (jeffkamo) [inline web/app/containers/checkout-confirmation/actions.js]: No it isn't. Got left over from when the vars weren't being used.

File: web/app/containers/checkout-confirmation/parsers/checkout-confirmation.js (hunk 2 above)
Comment (jnurse) [inline web/app/containers/checkout-confirmation/parsers/checkout-confirmation.js]: Yes it's still required.

File: web/app/app-provider.jsx [hunk 5]
```diff
@@ -26,16 +26,14 @@ const AppProvider = ({store}) => (
                 <Route component={PDP} path="checkout/cart/configure/id/*/product_id/*/" routeName="cartEditPage" />
                 <Route component={PDP} path="*.html" routeName="productDetailsPage" />
                 <Route component={CheckoutShipping} path="checkout/" routeName="checkingShipping" Header={CheckoutHeader} Footer={CheckoutFooter} />
+                {/*
```
Comment (jeremywiebe) [inline web/app/app-provider.jsx]: Nice explanation!
